### DISCUSSION TITLE
What should I do when fetch params (e.g. token) in getServerSideProps is changed?

### DISCUSSION THREAD

**Discussion by irekrog** [urql-graphql/urql#2344] [Q&A] [answered]
Hi, 

on the index page in `getServerSideProps` I'm fetching user data. There is an error because I have not a token from cookie. After click "go to second page" I'm creating a cookie and set its value and go to second page. On the second page I have a correct token and I'm doing the same things like on the first page (in `getServerSideProps` fetching user data but now with correct token). But there is still error (because of cache from previous page?) until refresh a page.

So what should I do to get correct data on the second page without refresh a page?

Thanks! :)

repo: https://github.com/irekrog/urql-test
recorded video:

https://user-images.githubusercontent.com/11172548/158050545-94d2b4a6-bfbc-4d6d-b376-e0f35c3daefd.mov

**Comment by JoviDeCroock** [comment, marked as answer]
That's an interesting one, the initUrqlClient method holds a scoped ref to the client so when using automatic ssr it can carry over between prepass and renderToString. In here that shoots us in the foot, which gives us two options, we async import initUrqlClient in your getXProps method by means of `const { initUrqlClient } = await import('next-urql')` or you leverage the `resetUrqlClient` import at the start of your method.

We can't really remove the scoped ref as that would probably cause some issues for folks using the normalized cache. This might be a change in how next bootstraps the ssr-runtime as this didn't use to be the case afaik, I should probably check if that next version is also doing that for automatic ssr

**Reply by irekrog** [reply]
Thanks for response! :) 

Ok, so I have added `resetUrlClient` on "login" and "logout" click -> https://github.com/irekrog/urql-test/commit/b432ea1fa3ecf865ac2153a621ff7a90ac21b0f8 and looks better now, but why in the network I have still 401 calls?

Option with `await import('next-urql')` did not help

https://user-images.githubusercontent.com/11172548/158061943-5f647097-4136-4a91-b62b-3ae862c2614c.mov
